cloud/ec2_client.go

```diff
@@ -1052,6 +1055,26 @@ func (c *awsClientImpl) AssumeRole(ctx context.Context, input *sts.AssumeRoleInp
 	return output, nil
 }
 
+// GetCallerIdentity is a wrapper for sts.GetCallerIdentity
+func (c *awsClientImpl) GetCallerIdentity(ctx context.Context, input *sts.GetCallerIdentityInput) (*sts.GetCallerIdentityOutput, error) {
+	var output *sts.GetCallerIdentityOutput
+	var err error
+	err = utility.Retry(
+		ctx,
+		func() (bool, error) {
+			msg := makeAWSLogMessage("GetCallerIdentity", fmt.Sprintf("%T", c), input)
+			// GetCallerIdentity doesn't require permissions and we should retry on
```

Comment:
this says we _should_ retry on any error, but here we return false, which would cause us to not retry. Should we be returning true? Alternatively if we don't intend to retry, then we don't need the retry loop at all, we can just call GetCallerIdentity directly

Reply:
Added an if err != nil -> return true to retry on error and adjusted the comment to reflect the intention (retry on network errors)